### PR TITLE
add Error handle module

### PR DESCRIPTION
**Pull request by ovekyc**
There is route issue which api request flow to React route, we solve that

We implement traditional Express error Handler
 and
ErrorHandler class to deal the error status

Sorry for large chunked commit

### CODE REVIEW

File: src/server/ErrorHandler.js [hunk 1]
```diff
@@ -0,0 +1,8 @@
+export class APIError {
```
Comment (festinalente91) [inline src/server/ErrorHandler.js:1]: We have to discuss about this class later 
LGTM

**Comment by festinalente91** [comment]
LGTM seems only work outside :(

**Comment by hewonjeong** [comment]
LGTM

File: src/server/routes/api-routes.js [hunk 2]
```diff
@@ -5,4 +5,24 @@ const router = express.Router();
...
 router.use('/items', items);
 
+router.use('/*', (req, res) => {
+  if (res.statusCode === 200 || res.statusCode === 304) {
+    return;
+  }
+  res.status(404).send('API Request > 404 - Page Not Found');
+});
+
+router.use((errHandler, req, res) => {
```
Comment (HewonJeong) [inline src/server/routes/api-routes.js]: I heard error handler function in Express requires 4 args(err, req, res, next).
this function will work in error situations like 400 or 500 in our item api controller?

File: src/server/routes/api-routes.js [hunk 3]
```diff
@@ -5,4 +5,24 @@ const router = express.Router();
...
+  res.status(errHandler.statusCode).send({
+    error: errHandler.message
+  });
+});
```
Comment (hewonjeong) [inline src/server/routes/api-routes.js:20]: I heard error handler function in Express requires 4 args(err, req, res, next).
this function will work in error situations like 400 or 500 in our item api controller?

File: src/server/routes/api-routes.js [hunk 4]
```diff
@@ -5,4 +5,24 @@ const router = express.Router();
 
 router.use('/items', items);
 
+router.use('/*', (req, res) => {
+  if (res.statusCode === 200 || res.statusCode === 304) {
+    return;
+  }
+  res.status(404).send('API Request > 404 - Page Not Found');
+});
+
```
Comment (hewonjeong) [inline src/server/routes/api-routes.js:14]: image
If I generate 400 error(Item not found error of GetById func), this func will be call but call other callback so occur some errors.
How can we solve itT_T

Comment (ovekyc) [reply src/server/routes/api-routes.js:14]: Is that one at this branch?? It seems like current master branch 
We just solve that in this PR 
Please check it out again

File: src/server/routes/api-routes.js (hunk 3 above)
Comment (ovekyc) [reply src/server/routes/api-routes.js:20]: Yes you are right.
But we dont use 4th arg , next
Unused variable is recognized Error in ESlint
So i remove it

**Comment by festinalente91** [comment]
LGTM

File: src/server/routes/api-routes.js [hunk 5]
```diff
@@ -5,4 +5,25 @@ const router = express.Router();
...
+router.use((errHandler, req, res, next) => {
+  res.status(errHandler.statusCode).send({
+    error: errHandler.message
+  });
+  next(errHandler.error);
```
Comment (ovekyc) [inline src/server/routes/api-routes.js:19]: The next is 'express's default Error handler

File: src/server/routes/api-routes.js [hunk 6]
```diff
@@ -5,4 +5,25 @@ const router = express.Router();
...
 router.use('/items', items);
 
+router.use('/*', (req, res) => {
+  if (res.statusCode === 200 || res.statusCode === 304) {
+    return;
+  }
+  res.status(404).send('API Request > 404 - Page Not Found');
+});
+
+router.use((errHandler, req, res, next) => {
```
Comment (ovekyc) [inline src/server/routes/api-routes.js:15]: We should make error handler like that form
4 args are strictly required by express

**Comment by ovekyc** [comment]
LGTM

**Comment by festinalente91** [comment]
LGTM

**Pull request merged**